### PR TITLE
Cleanup build and packaging

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR does a first round of cleanup after the build and packaging split

### Motivation

Reduce code duplication and sprinkling of various environment variables throughout our jobs

### CODE REVIEW

File: .gitlab/packaging/deb.yml [hunk 1]
```diff
@@ -22,108 +22,95 @@
   cache:
     - !reference [.cache_omnibus_ruby_deps, cache]
 
+.package_deb_x86:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["arch:amd64"]
+  variables:
+    DD_PKG_ARCH: "x86_64"
+    PACKAGE_ARCH: amd64
```
Comment (amenasria) [inline .gitlab/packaging/deb.yml:30]: **❓ question:** ‏I fear that later someone might forget the `"` while modifying this, isn't it possible to get rid of them everywhere ?

**Review by amenasria — APPROVED** [review]
LGTM as long as the CI passes !

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**